WalletWasabi/WabiSabi/Backend/Rounds/CoinJoinStorage/CoinJoinIdStore.cs

```diff
@@ -0,0 +1,63 @@
+using NBitcoin;
+using System.IO;
+using System.Linq;
+using WalletWasabi.Logging;
+
+namespace WalletWasabi.WabiSabi.Backend.Rounds.CoinJoinStorage;
+
+public class CoinJoinIdStore : ICoinJoinIdStore
+{
+	private InMemoryCoinJoinIdStore InMemoryCoinJoinIdStore { get; set; }
+	public string CoinJoinIdStoreFilePath { get; set; }
+	private object FileWriteLock { get; set; } = new();
+
+	public CoinJoinIdStore() : this(string.Empty)
+	{
+	}
+
+	public CoinJoinIdStore(string coinJoinIdStoreFilePath)
+	{
+		CoinJoinIdStoreFilePath = coinJoinIdStoreFilePath;
+		InMemoryCoinJoinIdStore = InMemoryCoinJoinIdStore.LoadFromFile(coinJoinIdStoreFilePath);
+	}
+
+	public void Append(uint256 id)
+	{
+		InMemoryCoinJoinIdStore.Add(id);
```

Comment:
Make sure If it is already added do not append it again to the file. 
InMemoryCoinJoinIdStore.Add could be TryAdd so it would return that if it was already added or not.